### PR TITLE
D3ASIM-2927: Fixed parsing of the live data device info from the MQTT…

### PR DESCRIPTION
**Pull request by spyrostz**
… subscriber. Do not read live data for Canary Networks that are not running. Fixed decoding in received MQTT data.

### CODE REVIEW

**Review by faizan2590 — APPROVED** [review]
LGTM

File: d3a_api_client/utils.py [hunk 1]
```diff
@@ -197,6 +197,7 @@ def list_running_canary_networks_and_devices_with_live_data(domain_name):
     data = endpoint(query=query)
 
     return {
-        cn["uuid"]: json.loads(cn["scenarioData"]["forecastStreamAreaMapping"])
+        cn["uuid"]: cn["scenarioData"]["forecastStreamAreaMapping"]
         for cn in data["data"]["listCanaryNetworks"]["configurations"]
-    }
+        if cn["resultsStatus"] == "running"
```
Comment (hannesdiedrich) [inline d3a_api_client/utils.py:202]: `else None` is not needed here?

**Review by hannesdiedrich — APPROVED** [review]
LGTM

File: d3a_api_client/utils.py (hunk 1 above)
Comment (spyrostz) [reply d3a_api_client/utils.py:202]: No, if the Canary Network is not running then we simply do not connect to it when the MQTT subscriber starts. When the CN starts the MQTT subscriber will be notified via the Redis queue.

**Pull request merged**